### ISSUE TITLE
Falso positive 'deprecated-method' for time.perf_counter() on Python 3.6 and 3.7

### ISSUE THREAD

**Issue by andy-maier** [PyCQA/pylint#4966]
### Bug description

```python
#!/usr/bin/env python
# File: pylint_perf_counter.py
"""
Demonstrate the false positive 'deprecated-method' in Pylint 2.10.2.
"""

from __future__ import print_function
try:
    # time.perf_counter() was added in Python 3.3
    from time import perf_counter
except ImportError:
    # time.clock() was deprecated in Python 3.3
    from time import clock as perf_counter

print(perf_counter)

# For the following line, Pylint 2.10.2 raises deprecated-method on Python 3.6
# and 3.7, but not on Python 3.8 and 3.9. Did not try Python 3.5 and below.
print(perf_counter())
```

### Command used

```shell
pylint pylint_perf_counter.py  # with no config file present
# on Python versions 3.7 and 3.9
```

### Pylint output

```shell
On Python 3.7:
************* Module pylint_perf_counter
pylint_perf_counter.py:19:6: W1505: Using deprecated method perf_counter() (deprecated-method)

On Python 3.9:
(no issues reported)
```

### Expected behavior

The code uses time.perf_counter() on Python >=3.3, and since that function is not deprecated, Pylint should not raise "deprecated-method" on Python 3.7 and below. Note that on Python 3.8 and above, Pylint does not raise that issue.

Maybe Pylint gets confused about the except clause where the deprecated time.clock() is imported. Changing that to a non-deprecated function causes the "deprecated-method" issue to no longer be raised.

### Pylint version

```shell
On Python 3.7:
pylint 2.10.2
astroid 2.7.2
Python 3.7.11 (default, Jul  6 2021, 12:42:50) 
[Clang 12.0.0 (clang-1200.0.32.29)]

On Python 3.9:
pylint 2.10.2
astroid 2.7.2
Python 3.9.6 (default, Jun 29 2021, 06:20:32) 
[Clang 12.0.0 (clang-1200.0.32.29)]
```

### OS / Environment

macOs 10.15.7

### Additional dependencies

Dependencies of Pylint:

On Python 3.7:
$ pipdeptree --python $(which python) -p pylint
pylint==2.10.2
  - astroid [required: >=2.7.2,<2.8, installed: 2.7.3]
    - lazy-object-proxy [required: >=1.4.0, installed: 1.6.0]
    - setuptools [required: >=20.0, installed: 57.5.0]
    - typed-ast [required: >=1.4.0,<1.5, installed: 1.4.3]
    - typing-extensions [required: >=3.7.4, installed: 3.10.0.2]
    - wrapt [required: >=1.11,<1.13, installed: 1.12.1]
  - isort [required: >=4.2.5,<6, installed: 5.9.3]
  - mccabe [required: >=0.6,<0.7, installed: 0.6.1]
  - platformdirs [required: >=2.2.0, installed: 2.3.0]
  - toml [required: >=0.7.1, installed: 0.10.2]

On Python 3.9:
$ pipdeptree --python $(which python) -p pylint
pylint==2.10.2
  - astroid [required: >=2.7.2,<2.8, installed: 2.7.3]
    - lazy-object-proxy [required: >=1.4.0, installed: 1.6.0]
    - setuptools [required: >=20.0, installed: 57.5.0]
    - wrapt [required: >=1.11,<1.13, installed: 1.12.1]
  - isort [required: >=4.2.5,<6, installed: 5.9.3]
  - mccabe [required: >=0.6,<0.7, installed: 0.6.1]
  - platformdirs [required: >=2.2.0, installed: 2.3.0]
  - toml [required: >=0.7.1, installed: 0.10.2]

Note that pipdeptree does not report typed-ast and typing-extensions as dependencies of astroid on Python 3.9. They are both not installed on Python 3.9. Did not dig into the details of that.

**Comment by jacobtylerwalls** [comment]
We consciously decided to warn on all versions for any deprecated usages, see https://github.com/pylint-dev/pylint/issues/9489
